tensorbay/client/version.py

```diff
@@ -659,3 +660,91 @@ def _generate_jobs(
             The generator of job info.
 
         """
+
+    def create_squash_and_merge_job(
+        self,
+        title: str = "",
+        description: str = "",
+        *,
+        draft_title: str,
+        source_branch_name: str,
+        target_branch_name: Optional[str] = None,
+        draft_description: str = "",
+        strategy: Optional[str] = "abort",
+    ) -> SquashAndMergeJob:
+        """Create a :class:`SquashAndMergeJob`.
+
+        Squash commits in source branch, then merge into target branch by creating a new draft.
+        If the target branch name is not given, the draft will be based on the branch name stored
+        in the dataset client. And during merging, the conflicts between branches can be resolved
+        in three different strategies: "abort", "override" and "skip".
+
+        Arguments:
+            title: The SquashAndMergeJob title.
+            description: The SquashAndMergeJob description.
+            draft_title: The draft title.
+            source_branch_name: The name of the branch to be squashed.
+            target_branch_name: The target branch name of the merge operation.
+            draft_description: The draft description.
+            strategy: The strategy of handling the branch conflict. There are three options:
+
+                1. "abort": abort the opetation;
+                2. "override": the squashed branch will override the target branch;
+                3. "skip": keep the origin branch.
+
+        Raises:# flake8: noqa: F402
+            StatusError: When squashing and merging without basing on a branch.
+
+        Return:
+            The SquashAndMergeJob.
+
+        """
+
+    def get_squash_and_merge_job(self, job_id: str) -> SquashAndMergeJob:
+        """Get a :class:`SquashAndMergeJob`.
+
+        Arguments:
+            job_id: The SquashAndMergeJob id.
+
+        Return:
+            The SquashAndMergeJob.
+
+        """
+
+    def list_squash_and_merge_jobs(
+        self, status: Optional[str] = None
+    ) -> PagingList[SquashAndMergeJob]:
+        """List the SquashAndMergeJob.
+
+        Arguments:
+            status: The SquashAndMergeJob status which includes "QUEUING", "PROCESSING", "SUCCESS",
+                    "FAIL", "ABORT" and None. None means all kinds of status.
+
+        Return:
+            The PagingList of SquashAndMergeJob.
+
+        """
+
+    def abort_squash_and_merge_job(self, job_id: str) -> None:
+        """Abort a :class:`SquashAndMergeJob`.
+
+        Arguments:
+            job_id: The SquashAndMergeJob id.
+
+        """
+
+    def retry_squash_and_merge_job(self, job_id: str) -> None:
+        """Retry a :class:`SquashAndMergeJob`.
+
+        Arguments:
+            job_id: The SquashAndMergeJob id.
+
+        """
+
+    def delete_squash_and_merge_job(self, job_id: str) -> None:
```

Comment:
we have different kinds of operators for one job: get, list, delete, abort, retry etc. and we put them all in `JobMixin` then put then all in `DatasetClient`.
If we have three or more job types, there will be too many methods.
Which will pollute the `DatasetClient` which makes users hard to find specified methods.

Reply:
But when user want to create sth, he should input `create_` first, then use `tab` to see what methods are available.

Reply:
can we separate different type of jobs into different layer?